src/arena.c

```diff
@@ -2054,9 +2054,11 @@ arena_new(tsdn_t *tsdn, unsigned ind, extent_hooks_t *extent_hooks) {
 		/* background_thread_create() handles reentrancy internally. */
 		if (have_background_thread) {
 			bool err;
+			malloc_mutex_unlock(tsdn, &arenas_lock);
```

Comment:
Actually ```arena_set(ind, arena);``` is done before this (I changed that as part of a previous refactor). After we return from this function, we pretty much unlock immediately.